### PR TITLE
Created post love api

### PR DESCRIPTION
**Pull request by alammoiz**

### CODE REVIEW

File: routes/post.js [hunk 1]
```diff
@@ -21,6 +21,14 @@ router.post(
   postController.createComment
 );
 
+router.post(
+  "/:postId/love",
+  tokenMiddleware.checkToken,
+  postController.validate("createPost"),
```
Comment (Ananto30) [inline routes/post.js:27]: I think we don't need any validator here. We are just taking no `body` in this POST request, right?

File: yarn.lock [hunk 2]
```diff
@@ -0,0 +1,2982 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (Ananto30) [inline yarn.lock:1]: Please remove this file 🙏 We already got the package-lock.json

**Review by Ananto30 — CHANGES_REQUESTED** [review]

File: routes/post.js (hunk 1 above)
Comment (alammoiz) [reply routes/post.js]: Yes you're right, Pushed again @Ananto30

File: yarn.lock (hunk 2 above)
Comment (alammoiz) [reply yarn.lock]: Sorry for that, Removed now

File: controller/post.controller.js [hunk 3]
```diff
@@ -68,6 +68,29 @@ exports.createComment = async (req, res) => {
...
 
+
+exports.reactLove = async (req, res) => {
+  try {
+    const errors = validationResult(req);
+
+    if (!errors.isEmpty()) {
+      res.status(422).json({ errors: errors.array() });
+      return;
+    }
```
Comment (Ananto30) [inline controller/post.controller.js:79]: As we don't need validation, we need to remove this too.

File: service/post.service.js [hunk 4]
```diff
@@ -74,6 +75,34 @@ exports.createComment = async (author, text, postId) => {
...
+    },
+    {
+      new: true,
+    }
+  );
+
+  const data = {
+    username: author,
+    summary: `reacted love on ${post.author}'s post`,
+    extraText: text.substring(0, 50),
```
Comment (Ananto30) [inline service/post.service.js:98]: As there is no `body` in the request we cannot get the `text`, please remove it.

File: service/post.service.js [hunk 5]
```diff
@@ -74,6 +75,34 @@ exports.createComment = async (author, text, postId) => {
   return post;
 };
 
+exports.reactLove = async (author, postId) => {
+  const post = await Post.findOneAndUpdate(
```
Comment (Ananto30) [inline service/post.service.js:79]: If we update in this manner, people can spam it. We need to check if the user already love it or not, if already loved, don't update it, if not already loved, update it.

See what can be wrong 😅
image

**Review by Ananto30 — CHANGES_REQUESTED** [review]

**Comment by Ananto30** [comment]
Awesome work! Thanks a lot 🙏

**Review by Ananto30 — APPROVED** [review]

**Pull request merged**